### PR TITLE
Add monitoring modules

### PR DESCRIPTION
**Pull request by JakeNeyer**
Starts work on [#3537](https://github.com/polytomic/app/issues/3537)

This PR introduces the following optional monitoring alarms via CloudWatch alarms and SNS:

|                       | ECS | RDS | Elasticache |   | Default Threshold |
|-----------------------|-----|-----|-------------|---|-------------------|
| CPUUtilization High   | ✅   | ✅   | ✅           |   | 80%               |
| MemoryUtilizationHigh | ✅   |     |             |   | 80%               |
| RunningTasksCount     | ✅   |     |             |   | <1                |
| CurrentConnections    |     |     | ✅           |   | >1000             |
| FreeableMemoryLow     |     | ✅   | ✅           |   | 64MB              |
| BurstBalanceLow       |     | ✅   |             |   | 20                |
| CPUCreditBalanceLow   |     | ✅   |             |   | 5                 |
| DiskQueueDepthHigh    |     | ✅   |             |   | 64                |
| SwapUsageHigh         |     | ✅   |             |   | 256MB             |
| FreeStorageSpaceLow   |     | ✅   |             |   | 2GB               |

Subscribers to the SNS topic are added via email by adding them to `alert_emails`

e.g. `  alert_emails = ["user@example.com"]`

### CODE REVIEW

File: terraform/modules/monitoring/rds-alerts/main.tf [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    aws_sns_topic_policy.default
+  ]
+}
+
+resource "aws_sns_topic_policy" "default" {
+  arn    = join("", var.sns_topic_arns)
+  policy = join("", data.aws_iam_policy_document.sns_topic_policy.*.json)
+}
+
+data "aws_iam_policy_document" "sns_topic_policy" {
```
Comment (JakeNeyer) [inline terraform/modules/monitoring/rds-alerts/main.tf:168]: I meant to pull this out of here and move it to the root module

File: terraform/modules/ecs/monitoring.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
+resource "aws_kms_key" "alerts" {}
+
+resource "aws_sns_topic" "alerts" {
+  name              = "${var.prefix}-alerts"
+  kms_master_key_id = aws_kms_key.alerts.arn
+}
+
+resource "aws_sns_topic_subscription" "alert_emails" {
+  for_each  = toset(var.alert_emails)
```
Comment (nyergler) [inline terraform/modules/ecs/monitoring.tf:9]: Can we make this default to `monitoring-{deployment id}@polytomic.com`?

**Review by nyergler — APPROVED** [review]
This looks great; I think it'll give us a lot more confidence that things are running smoothly.

File: terraform/modules/ecs/monitoring.tf (hunk 2 above)
Comment (JakeNeyer) [reply terraform/modules/ecs/monitoring.tf:9]: Of course! Adding that now

**Pull request merged**